### PR TITLE
USB: Fix Coverity issues

### LINKED ISSUE #14954: [Coverity CID :197461]Insecure data handling in /subsys/usb/usb_device.c
Static code scan issues seen in File: /subsys/usb/usb_device.c
 Category: Insecure data handling
 Function: usb_set_configuration
 Component: Other
 CID: 197461
 Please fix or provide comments to square it off in coverity in the link: https://scan9.coverity.com/reports.htm#v32951/p12996

### LINKED ISSUE #14957: [Coverity CID :197458]Insecure data handling in /subsys/usb/usb_device.c
Static code scan issues seen in File: /subsys/usb/usb_device.c
 Category: Insecure data handling
 Function: usb_set_interface
 Component: Other
 CID: 197458
 Please fix or provide comments to square it off in coverity in the link: https://scan9.coverity.com/reports.htm#v32951/p12996

### PR DESCRIPTION
**Pull request by finikorg**
Cherry-picked 2 patches from another PR to avoid conflicts

Fixes #14957
Fixes #14954

### CODE REVIEW

File: subsys/usb/usb_device.c [hunk 1]
```diff
@@ -461,6 +461,45 @@ static bool usb_get_descriptor(u16_t type_index, u16_t lang_id,
 	return found;
 }
 
+static bool set_endpoint(const struct usb_ep_descriptor *ep_desc)
```
Comment (jfischer-phytec-iot) [inline subsys/usb/usb_device.c]: The problem here maybe that controller+driver may operate in high speed mode (usb_dc_dw.c) and this information (FS or HS) is lacking.

File: subsys/usb/usb_device.c [hunk 2]
```diff
@@ -461,6 +461,45 @@ static bool usb_get_descriptor(u16_t type_index, u16_t lang_id,
...
+	ep_cfg.ep_mps = sys_le16_to_cpu(ep_desc->wMaxPacketSize);
+
+	if (ep_desc->bmAttributes > USB_DC_EP_INTERRUPT) {
+		return false;
+	}
+
+	ep_cfg.ep_type = ep_desc->bmAttributes;
+
+	switch (ep_cfg.ep_type) {
+	case USB_DC_EP_BULK:
```
Comment (jfischer-phytec-iot) [inline subsys/usb/usb_device.c]:
```
if (defined(DT_USBHS_MAXIMUM_SPEED)) {
		if (ep_cfg.ep_mps > USB_MAX_BULK_HS_MPS) {
			return false;
		}
} else {
		if (ep_cfg.ep_mps > USB_MAX_BULK_FS_MPS) {
			return false;
		}
}
```

?

Comment (finikorg) [reply subsys/usb/usb_device.c]: I propose to use it like in the **header**:
```
#if defined(DT_USBHS_MAXIMUM_SPEED)
#define USB_MAX_BULK_MPS 512
#else
...
#endif
```

So we have this `DT_USBHS_MAXIMUM_SPEED` defined in DTS to use HS?

Comment (finikorg) [reply subsys/usb/usb_device.c]: BTW, for stm32 it is `DT_USB_MAXIMUM_SPEED`

**Comment by nashif** [comment]
@finikorg any updates? This is blocking the release.

**Comment by finikorg** [comment]
@jfischer-phytec-iot I have added PR #15140 to handle `maximum-spedd` issue, after it is merged we may add speed check.

**Comment by finikorg** [comment]
> @finikorg any updates? This is blocking the release.

Rebased and fixed conflicts.

File: subsys/usb/usb_device.c (hunk 1 above)
Comment (finikorg) [reply subsys/usb/usb_device.c]: I will add HS support after my high speed PR is merged #15140

**Comment by finikorg** [comment]
@jfischer-phytec-iot I added offloading MPS check to usb device controller, depending on the mode it would check parameters. They need to add checking for MPS, I propose for now support only Full Speed.

**Comment by finikorg** [comment]
I would add also unit test but we are stuck with problem merging before release (following patch #14874)

**Comment by nashif** [comment]
closing due to alternative fix

**Pull request closed without merging**